pkg/controllers/dynakube/extension/eec/reconciler.go

```diff
@@ -70,3 +84,106 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 
 	return r.createOrUpdateStatefulset(ctx)
 }
+
+func (r *reconciler) reconcileTLSSecret(ctx context.Context) error {
+	query := k8ssecret.Query(r.client, r.client, log)
+
+	secret, err := query.Get(ctx, types.NamespacedName{
+		Name:      getSelfSignedTLSSecretName(r.dk.Name),
+		Namespace: r.dk.Namespace,
+	})
+
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+
+	if r.dk.Spec.Templates.ExtensionExecutionController.TlsRefName != "" {
+		if secret != nil {
+			return query.Delete(ctx, secret)
+		}
+
+		return nil
+	}
+
+	if k8serrors.IsNotFound(err) {
+		err = r.createOrUpdateTLSSecret(ctx)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	if secret != nil {
+		err = r.reconcileTLSSecretExpiration(ctx, secret)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *reconciler) createOrUpdateTLSSecret(ctx context.Context) error {
```

Comment:
also what I found, there is no unit-tests for reconcile secret method.

Reply:
[Unit tests added](https://github.com/Dynatrace/dynatrace-operator/pull/3778/commits/7c824e188f80a032a4823a0e0aabb88002f66b51)